
wkdir='/Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis'

cd /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/00_raw_fastq/00_cp/


cp ExtrNeg1_S61_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/ExtrNeg1_S61_L001_R1_001.fastq.gz
cp EXTRNEG_S73_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/EXTRNEG_S73_L001_R1_001.fastq.gz
cp EXTRPOS_S72_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/EXTRPOS_S72_L001_R1_001.fastq.gz
cp GW10_S2_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW10_S2_L001_R1_001.fastq.gz
cp GW11_S3_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW11_S3_L001_R1_001.fastq.gz
cp GW12_S4_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW12_S4_L001_R1_001.fastq.gz
cp GW13_S5_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW13_S5_L001_R1_001.fastq.gz
cp GW14_S6_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW14_S6_L001_R1_001.fastq.gz
cp GW15_S7_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW15_S7_L001_R1_001.fastq.gz
cp GW16_S74_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW16_S74_L001_R1_001.fastq.gz
cp GW17_S8_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW17_S8_L001_R1_001.fastq.gz
cp GW18_S9_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW18_S9_L001_R1_001.fastq.gz
cp GW19_S10_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW19_S10_L001_R1_001.fastq.gz
cp GW1_S64_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW1_S64_L001_R1_001.fastq.gz
cp GW20_S11_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW20_S11_L001_R1_001.fastq.gz
cp GW21_S12_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW21_S12_L001_R1_001.fastq.gz
cp GW22_S13_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW22_S13_L001_R1_001.fastq.gz
cp GW23_S14_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW23_S14_L001_R1_001.fastq.gz
cp GW24_S15_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW24_S15_L001_R1_001.fastq.gz
cp GW25_S16_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW25_S16_L001_R1_001.fastq.gz
cp GW26_S17_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW26_S17_L001_R1_001.fastq.gz
cp GW27_S18_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW27_S18_L001_R1_001.fastq.gz
cp GW28_S75_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW28_S75_L001_R1_001.fastq.gz
cp GW29_S19_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW29_S19_L001_R1_001.fastq.gz
cp GW2_S65_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW2_S65_L001_R1_001.fastq.gz
cp GW30_S20_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW30_S20_L001_R1_001.fastq.gz
cp GW31_S21_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW31_S21_L001_R1_001.fastq.gz
cp GW32_S22_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW32_S22_L001_R1_001.fastq.gz
cp GW33_S76_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW33_S76_L001_R1_001.fastq.gz
cp GW34_S23_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW34_S23_L001_R1_001.fastq.gz
cp GW35_S77_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW35_S77_L001_R1_001.fastq.gz
cp GW36_S24_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW36_S24_L001_R1_001.fastq.gz
cp GW37_S25_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW37_S25_L001_R1_001.fastq.gz
cp GW38_S26_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW38_S26_L001_R1_001.fastq.gz
cp GW39_S27_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW39_S27_L001_R1_001.fastq.gz
cp GW3_S66_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW3_S66_L001_R1_001.fastq.gz
cp GW40_S28_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW40_S28_L001_R1_001.fastq.gz
cp GW41_S29_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW41_S29_L001_R1_001.fastq.gz
cp GW42_S30_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW42_S30_L001_R1_001.fastq.gz
cp GW43_S31_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW43_S31_L001_R1_001.fastq.gz
cp GW44_S32_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW44_S32_L001_R1_001.fastq.gz
cp GW45_S33_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW45_S33_L001_R1_001.fastq.gz
cp GW46_S34_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW46_S34_L001_R1_001.fastq.gz
cp GW47_S35_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW47_S35_L001_R1_001.fastq.gz
cp GW48_S36_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW48_S36_L001_R1_001.fastq.gz
cp GW49_S37_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW49_S37_L001_R1_001.fastq.gz
cp GW4_S67_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW4_S67_L001_R1_001.fastq.gz
cp GW50_S38_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW50_S38_L001_R1_001.fastq.gz
cp GW51_S39_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW51_S39_L001_R1_001.fastq.gz
cp GW52_S40_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW52_S40_L001_R1_001.fastq.gz
cp GW53_S41_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW53_S41_L001_R1_001.fastq.gz
cp GW54_S42_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW54_S42_L001_R1_001.fastq.gz
cp GW55_S43_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW55_S43_L001_R1_001.fastq.gz
cp GW56_S44_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW56_S44_L001_R1_001.fastq.gz
cp GW57_S45_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW57_S45_L001_R1_001.fastq.gz
cp GW58_S46_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW58_S46_L001_R1_001.fastq.gz
cp GW59_S78_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW59_S78_L001_R1_001.fastq.gz
cp GW5_S68_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW5_S68_L001_R1_001.fastq.gz
cp GW60_S47_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW60_S47_L001_R1_001.fastq.gz
cp GW61_S48_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW61_S48_L001_R1_001.fastq.gz
cp GW62_S79_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW62_S79_L001_R1_001.fastq.gz
cp GW63_S49_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW63_S49_L001_R1_001.fastq.gz
cp GW64_S50_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW64_S50_L001_R1_001.fastq.gz
cp GW65_S51_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW65_S51_L001_R1_001.fastq.gz
cp GW66_S52_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW66_S52_L001_R1_001.fastq.gz
cp GW67_S53_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW67_S53_L001_R1_001.fastq.gz
cp GW68_S54_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW68_S54_L001_R1_001.fastq.gz
cp GW69_S55_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW69_S55_L001_R1_001.fastq.gz
cp GW6_S69_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW6_S69_L001_R1_001.fastq.gz
cp GW70_S56_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW70_S56_L001_R1_001.fastq.gz
cp GW71_S57_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW71_S57_L001_R1_001.fastq.gz
cp GW72_S58_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW72_S58_L001_R1_001.fastq.gz
cp GW73_S59_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW73_S59_L001_R1_001.fastq.gz
cp GW74_S60_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW74_S60_L001_R1_001.fastq.gz
cp GW7_S70_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW7_S70_L001_R1_001.fastq.gz
cp GW8_S71_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW8_S71_L001_R1_001.fastq.gz
cp GW9_S1_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/GW9_S1_L001_R1_001.fastq.gz
cp PCRNeg1_S63_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/PCRNeg1_S63_L001_R1_001.fastq.gz
cp PCRNeg2_S81_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/PCRNeg2_S81_L001_R1_001.fastq.gz
cp PCRPos1_S62_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/PCRPos1_S62_L001_R1_001.fastq.gz
cp PCRPos2_S80_L001_R1_001.fastq.gz /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/PCRPos2_S80_L001_R1_001.fastq.gz


#copy dirksen forward reads to folder

cd /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/dirksen_reads/reads_fastq/

crap='_1.fastq.gz'

while read F; do
  cd $F
  cp $F$crap /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/$F$crap
  cd /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/dirksen_reads/reads_fastq/
done < /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/dirksen_reads/dirksen_acc_list.txt


#okay, the samuel reads

cd /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/samuel_reads/sra

for i in *.fastq; do gzip $i; done

for i in *.fastq.gz; do mv $i /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/$i; done

#qiime2 is picky
mv ERR1307271_1.fastq.gz ERR1307271_1_L002_R1_001.fastq.gz
mv ERR1307272_1.fastq.gz ERR1307272_2_L002_R1_001.fastq.gz
mv ERR1307273_1.fastq.gz ERR1307273_3_L002_R1_001.fastq.gz
mv ERR1307274_1.fastq.gz ERR1307274_4_L002_R1_001.fastq.gz
mv ERR1307275_1.fastq.gz ERR1307275_5_L002_R1_001.fastq.gz
mv ERR1307276_1.fastq.gz ERR1307276_6_L002_R1_001.fastq.gz
mv ERR1307277_1.fastq.gz ERR1307277_7_L002_R1_001.fastq.gz
mv ERR1307278_1.fastq.gz ERR1307278_8_L002_R1_001.fastq.gz
mv ERR1307279_1.fastq.gz ERR1307279_9_L002_R1_001.fastq.gz
mv ERR1307280_1.fastq.gz ERR1307280_10_L002_R1_001.fastq.gz
mv ERR1307281_1.fastq.gz ERR1307281_11_L002_R1_001.fastq.gz
mv ERR1307282_1.fastq.gz ERR1307282_12_L002_R1_001.fastq.gz
mv ERR1307283_1.fastq.gz ERR1307283_13_L002_R1_001.fastq.gz
mv ERR1307284_1.fastq.gz ERR1307284_14_L002_R1_001.fastq.gz
mv ERR1307285_1.fastq.gz ERR1307285_15_L002_R1_001.fastq.gz
mv ERR1307286_1.fastq.gz ERR1307286_16_L002_R1_001.fastq.gz
mv ERR1307287_1.fastq.gz ERR1307287_17_L002_R1_001.fastq.gz
mv ERR1307288_1.fastq.gz ERR1307288_18_L002_R1_001.fastq.gz
mv ERR1307290_1.fastq.gz ERR1307290_19_L002_R1_001.fastq.gz
mv ERR1307291_1.fastq.gz ERR1307291_20_L002_R1_001.fastq.gz
mv ERR1307292_1.fastq.gz ERR1307292_21_L002_R1_001.fastq.gz
mv ERR1307293_1.fastq.gz ERR1307293_22_L002_R1_001.fastq.gz
mv ERR1307294_1.fastq.gz ERR1307294_23_L002_R1_001.fastq.gz
mv ERR1307295_1.fastq.gz ERR1307295_24_L002_R1_001.fastq.gz
mv ERR1307296_1.fastq.gz ERR1307296_25_L002_R1_001.fastq.gz
mv ERR1307297_1.fastq.gz ERR1307297_26_L002_R1_001.fastq.gz
mv ERR1307298_1.fastq.gz ERR1307298_27_L002_R1_001.fastq.gz
mv ERR1307299_1.fastq.gz ERR1307299_28_L002_R1_001.fastq.gz
mv ERR1307300_1.fastq.gz ERR1307300_29_L002_R1_001.fastq.gz
mv ERR1307301_1.fastq.gz ERR1307301_30_L002_R1_001.fastq.gz
mv ERR1307302_1.fastq.gz ERR1307302_31_L002_R1_001.fastq.gz
mv ERR1307303_1.fastq.gz ERR1307303_32_L002_R1_001.fastq.gz
mv ERR1307304_1.fastq.gz ERR1307304_33_L002_R1_001.fastq.gz
mv ERR1307305_1.fastq.gz ERR1307305_34_L002_R1_001.fastq.gz
mv ERR1307306_1.fastq.gz ERR1307306_35_L002_R1_001.fastq.gz
mv ERR1307307_1.fastq.gz ERR1307307_36_L002_R1_001.fastq.gz
mv ERR1307308_1.fastq.gz ERR1307308_37_L002_R1_001.fastq.gz
mv ERR1307309_1.fastq.gz ERR1307309_38_L002_R1_001.fastq.gz
mv ERR1307310_1.fastq.gz ERR1307310_39_L002_R1_001.fastq.gz
mv ERR1307311_1.fastq.gz ERR1307311_40_L002_R1_001.fastq.gz
mv ERR1307312_1.fastq.gz ERR1307312_41_L002_R1_001.fastq.gz
mv ERR1307313_1.fastq.gz ERR1307313_42_L002_R1_001.fastq.gz
mv ERR1307314_1.fastq.gz ERR1307314_43_L002_R1_001.fastq.gz
mv ERR1307315_1.fastq.gz ERR1307315_44_L002_R1_001.fastq.gz
mv ERR1307316_1.fastq.gz ERR1307316_45_L002_R1_001.fastq.gz
mv ERR1307317_1.fastq.gz ERR1307317_46_L002_R1_001.fastq.gz
mv ERR1307318_1.fastq.gz ERR1307318_47_L002_R1_001.fastq.gz
mv ERR1307319_1.fastq.gz ERR1307319_48_L002_R1_001.fastq.gz
mv ERR1307320_1.fastq.gz ERR1307320_49_L002_R1_001.fastq.gz
mv ERR1307321_1.fastq.gz ERR1307321_50_L002_R1_001.fastq.gz
mv ERR1307322_1.fastq.gz ERR1307322_51_L002_R1_001.fastq.gz
mv ERR1307323_1.fastq.gz ERR1307323_52_L002_R1_001.fastq.gz
mv ERR1307324_1.fastq.gz ERR1307324_53_L002_R1_001.fastq.gz
mv ERR1307325_1.fastq.gz ERR1307325_54_L002_R1_001.fastq.gz
mv ERR1307326_1.fastq.gz ERR1307326_55_L002_R1_001.fastq.gz
mv ERR1307327_1.fastq.gz ERR1307327_56_L002_R1_001.fastq.gz
mv ERR1307328_1.fastq.gz ERR1307328_57_L002_R1_001.fastq.gz
mv ERR1307329_1.fastq.gz ERR1307329_58_L002_R1_001.fastq.gz
mv ERR1307330_1.fastq.gz ERR1307330_59_L002_R1_001.fastq.gz
mv ERR1307331_1.fastq.gz ERR1307331_60_L002_R1_001.fastq.gz
mv ERR1307332_1.fastq.gz ERR1307332_61_L002_R1_001.fastq.gz
mv ERR1307333_1.fastq.gz ERR1307333_62_L002_R1_001.fastq.gz
mv ERR1307334_1.fastq.gz ERR1307334_63_L002_R1_001.fastq.gz
mv ERR1307335_1.fastq.gz ERR1307335_64_L002_R1_001.fastq.gz
mv ERR1307336_1.fastq.gz ERR1307336_65_L002_R1_001.fastq.gz
mv ERR1307337_1.fastq.gz ERR1307337_66_L002_R1_001.fastq.gz
mv ERR1307338_1.fastq.gz ERR1307338_67_L002_R1_001.fastq.gz
mv ERR1307339_1.fastq.gz ERR1307339_68_L002_R1_001.fastq.gz
mv ERR1307340_1.fastq.gz ERR1307340_69_L002_R1_001.fastq.gz
mv ERR1307341_1.fastq.gz ERR1307341_70_L002_R1_001.fastq.gz
mv ERR1307342_1.fastq.gz ERR1307342_71_L002_R1_001.fastq.gz
mv ERR1307343_1.fastq.gz ERR1307343_72_L002_R1_001.fastq.gz
mv ERR1307344_1.fastq.gz ERR1307344_73_L002_R1_001.fastq.gz
mv ERR1307345_1.fastq.gz ERR1307345_74_L002_R1_001.fastq.gz
mv ERR1307346_1.fastq.gz ERR1307346_75_L002_R1_001.fastq.gz
mv ERR1307347_1.fastq.gz ERR1307347_76_L002_R1_001.fastq.gz
mv ERR1307348_1.fastq.gz ERR1307348_77_L002_R1_001.fastq.gz
mv ERR1307349_1.fastq.gz ERR1307349_78_L002_R1_001.fastq.gz
mv ERR1307350_1.fastq.gz ERR1307350_79_L002_R1_001.fastq.gz
mv ERR1307351_1.fastq.gz ERR1307351_80_L002_R1_001.fastq.gz
mv ERR1307352_1.fastq.gz ERR1307352_81_L002_R1_001.fastq.gz
mv ERR1307353_1.fastq.gz ERR1307353_82_L002_R1_001.fastq.gz
mv ERR1307354_1.fastq.gz ERR1307354_83_L002_R1_001.fastq.gz
mv ERR1307355_1.fastq.gz ERR1307355_84_L002_R1_001.fastq.gz
mv ERR1307356_1.fastq.gz ERR1307356_85_L002_R1_001.fastq.gz
mv ERR1307357_1.fastq.gz ERR1307357_86_L002_R1_001.fastq.gz
mv ERR1307358_1.fastq.gz ERR1307358_87_L002_R1_001.fastq.gz
mv ERR1307359_1.fastq.gz ERR1307359_88_L002_R1_001.fastq.gz
mv ERR1307360_1.fastq.gz ERR1307360_89_L002_R1_001.fastq.gz
mv ERR1307361_1.fastq.gz ERR1307361_90_L002_R1_001.fastq.gz
mv ERR1307362_1.fastq.gz ERR1307362_91_L002_R1_001.fastq.gz
mv ERR1307363_1.fastq.gz ERR1307363_92_L002_R1_001.fastq.gz
mv ERR1307364_1.fastq.gz ERR1307364_93_L002_R1_001.fastq.gz
mv ERR1307365_1.fastq.gz ERR1307365_94_L002_R1_001.fastq.gz
mv ERR1307366_1.fastq.gz ERR1307366_95_L002_R1_001.fastq.gz
mv ERR1307367_1.fastq.gz ERR1307367_96_L002_R1_001.fastq.gz
mv ERR1307368_1.fastq.gz ERR1307368_97_L002_R1_001.fastq.gz
mv ERR1307369_1.fastq.gz ERR1307369_98_L002_R1_001.fastq.gz
mv ERR1307370_1.fastq.gz ERR1307370_99_L002_R1_001.fastq.gz
mv ERR1307371_1.fastq.gz ERR1307371_100_L002_R1_001.fastq.gz
mv ERR1307372_1.fastq.gz ERR1307372_101_L002_R1_001.fastq.gz
mv ERR1307373_1.fastq.gz ERR1307373_102_L002_R1_001.fastq.gz
mv ERR1307374_1.fastq.gz ERR1307374_103_L002_R1_001.fastq.gz
mv ERR1307375_1.fastq.gz ERR1307375_104_L002_R1_001.fastq.gz
mv ERR1307376_1.fastq.gz ERR1307376_105_L002_R1_001.fastq.gz
mv ERR1307377_1.fastq.gz ERR1307377_106_L002_R1_001.fastq.gz
mv ERR1307378_1.fastq.gz ERR1307378_107_L002_R1_001.fastq.gz
mv ERR1307379_1.fastq.gz ERR1307379_108_L002_R1_001.fastq.gz
mv ERR1307380_1.fastq.gz ERR1307380_109_L002_R1_001.fastq.gz
mv ERR1307381_1.fastq.gz ERR1307381_110_L002_R1_001.fastq.gz
mv ERR1307382_1.fastq.gz ERR1307382_111_L002_R1_001.fastq.gz
mv ERR1307383_1.fastq.gz ERR1307383_112_L002_R1_001.fastq.gz
mv ERR1307384_1.fastq.gz ERR1307384_113_L002_R1_001.fastq.gz
mv ERR1307385_1.fastq.gz ERR1307385_114_L002_R1_001.fastq.gz
mv ERR1307386_1.fastq.gz ERR1307386_115_L002_R1_001.fastq.gz
mv ERR1307387_1.fastq.gz ERR1307387_116_L002_R1_001.fastq.gz
mv ERR1307388_1.fastq.gz ERR1307388_117_L002_R1_001.fastq.gz
mv ERR1307389_1.fastq.gz ERR1307389_118_L002_R1_001.fastq.gz
mv ERR1307390_1.fastq.gz ERR1307390_119_L002_R1_001.fastq.gz
mv ERR1307391_1.fastq.gz ERR1307391_120_L002_R1_001.fastq.gz
mv ERR1307392_1.fastq.gz ERR1307392_121_L002_R1_001.fastq.gz
mv ERR1307393_1.fastq.gz ERR1307393_122_L002_R1_001.fastq.gz
mv ERR1307394_1.fastq.gz ERR1307394_123_L002_R1_001.fastq.gz
mv ERR1307395_1.fastq.gz ERR1307395_124_L002_R1_001.fastq.gz
mv ERR1307396_1.fastq.gz ERR1307396_125_L002_R1_001.fastq.gz
mv ERR1307397_1.fastq.gz ERR1307397_126_L002_R1_001.fastq.gz
mv ERR1307398_1.fastq.gz ERR1307398_127_L002_R1_001.fastq.gz
mv ERR1307399_1.fastq.gz ERR1307399_128_L002_R1_001.fastq.gz
mv ERR1307400_1.fastq.gz ERR1307400_129_L002_R1_001.fastq.gz
mv ERR1307401_1.fastq.gz ERR1307401_130_L002_R1_001.fastq.gz
mv ERR1307402_1.fastq.gz ERR1307402_131_L002_R1_001.fastq.gz
mv ERR1307403_1.fastq.gz ERR1307403_132_L002_R1_001.fastq.gz
mv ERR1307404_1.fastq.gz ERR1307404_133_L002_R1_001.fastq.gz
mv ERR1307405_1.fastq.gz ERR1307405_134_L002_R1_001.fastq.gz
mv ERR1307406_1.fastq.gz ERR1307406_135_L002_R1_001.fastq.gz
mv ERR1307407_1.fastq.gz ERR1307407_136_L002_R1_001.fastq.gz
mv ERR1307408_1.fastq.gz ERR1307408_137_L002_R1_001.fastq.gz
mv ERR1307409_1.fastq.gz ERR1307409_138_L002_R1_001.fastq.gz
mv ERR1307410_1.fastq.gz ERR1307410_139_L002_R1_001.fastq.gz
mv ERR1307411_1.fastq.gz ERR1307411_140_L002_R1_001.fastq.gz
mv ERR1307412_1.fastq.gz ERR1307412_141_L002_R1_001.fastq.gz
mv ERR1307413_1.fastq.gz ERR1307413_142_L002_R1_001.fastq.gz
mv ERR1307414_1.fastq.gz ERR1307414_143_L002_R1_001.fastq.gz
mv ERR1307415_1.fastq.gz ERR1307415_144_L002_R1_001.fastq.gz
mv ERR1307416_1.fastq.gz ERR1307416_145_L002_R1_001.fastq.gz
mv ERR1307417_1.fastq.gz ERR1307417_146_L002_R1_001.fastq.gz
mv ERR1307418_1.fastq.gz ERR1307418_147_L002_R1_001.fastq.gz
mv ERR1307419_1.fastq.gz ERR1307419_148_L002_R1_001.fastq.gz
mv ERR1307420_1.fastq.gz ERR1307420_149_L002_R1_001.fastq.gz
mv ERR1307421_1.fastq.gz ERR1307421_150_L002_R1_001.fastq.gz
mv ERR1307422_1.fastq.gz ERR1307422_151_L002_R1_001.fastq.gz
mv ERR1307423_1.fastq.gz ERR1307423_152_L002_R1_001.fastq.gz
mv ERR1307424_1.fastq.gz ERR1307424_153_L002_R1_001.fastq.gz
mv ERR1307425_1.fastq.gz ERR1307425_154_L002_R1_001.fastq.gz
mv ERR1307426_1.fastq.gz ERR1307426_155_L002_R1_001.fastq.gz
mv ERR1307427_1.fastq.gz ERR1307427_156_L002_R1_001.fastq.gz
mv ERR1307428_1.fastq.gz ERR1307428_157_L002_R1_001.fastq.gz
mv ERR1307429_1.fastq.gz ERR1307429_158_L002_R1_001.fastq.gz
mv ERR1307430_1.fastq.gz ERR1307430_159_L002_R1_001.fastq.gz
mv ERR1307431_1.fastq.gz ERR1307431_160_L002_R1_001.fastq.gz
mv ERR1307432_1.fastq.gz ERR1307432_161_L002_R1_001.fastq.gz
mv ERR1307433_1.fastq.gz ERR1307433_162_L002_R1_001.fastq.gz
mv ERR1307434_1.fastq.gz ERR1307434_163_L002_R1_001.fastq.gz
mv ERR1307435_1.fastq.gz ERR1307435_164_L002_R1_001.fastq.gz
mv ERR1307436_1.fastq.gz ERR1307436_165_L002_R1_001.fastq.gz
mv ERR1307437_1.fastq.gz ERR1307437_166_L002_R1_001.fastq.gz
mv ERR1307438_1.fastq.gz ERR1307438_167_L002_R1_001.fastq.gz
mv ERR1307439_1.fastq.gz ERR1307439_168_L002_R1_001.fastq.gz
mv ERR1307440_1.fastq.gz ERR1307440_169_L002_R1_001.fastq.gz
mv ERR1307441_1.fastq.gz ERR1307441_170_L002_R1_001.fastq.gz
mv ERR1307442_1.fastq.gz ERR1307442_171_L002_R1_001.fastq.gz
mv ERR1307443_1.fastq.gz ERR1307443_172_L002_R1_001.fastq.gz
mv ERR1307444_1.fastq.gz ERR1307444_173_L002_R1_001.fastq.gz
mv ERR1307445_1.fastq.gz ERR1307445_174_L002_R1_001.fastq.gz
mv ERR1307446_1.fastq.gz ERR1307446_175_L002_R1_001.fastq.gz
mv ERR1307447_1.fastq.gz ERR1307447_176_L002_R1_001.fastq.gz
mv ERR1307448_1.fastq.gz ERR1307448_177_L002_R1_001.fastq.gz
mv ERR1307449_1.fastq.gz ERR1307449_178_L002_R1_001.fastq.gz
mv ERR1307450_1.fastq.gz ERR1307450_179_L002_R1_001.fastq.gz
mv ERR1307451_1.fastq.gz ERR1307451_180_L002_R1_001.fastq.gz 

mv SRR5094338.fastq.gz SRR5094338_1_L003_R1_001.fastq.gz
mv SRR5094346.fastq.gz SRR5094346_2_L003_R1_001.fastq.gz
mv SRR5094349.fastq.gz SRR5094349_3_L003_R1_001.fastq.gz
mv SRR5094351.fastq.gz SRR5094351_4_L003_R1_001.fastq.gz
mv SRR5094352.fastq.gz SRR5094352_5_L003_R1_001.fastq.gz
mv SRR5094356.fastq.gz SRR5094356_6_L003_R1_001.fastq.gz
mv SRR5094361.fastq.gz SRR5094361_7_L003_R1_001.fastq.gz
mv SRR5094362.fastq.gz SRR5094362_8_L003_R1_001.fastq.gz
mv SRR5094366.fastq.gz SRR5094366_9_L003_R1_001.fastq.gz
mv SRR5094370.fastq.gz SRR5094370_10_L003_R1_001.fastq.gz
mv SRR5094374.fastq.gz SRR5094374_11_L003_R1_001.fastq.gz
mv SRR5094376.fastq.gz SRR5094376_12_L003_R1_001.fastq.gz
mv SRR5094384.fastq.gz SRR5094384_13_L003_R1_001.fastq.gz
mv SRR5094388.fastq.gz SRR5094388_14_L003_R1_001.fastq.gz
mv SRR5094389.fastq.gz SRR5094389_15_L003_R1_001.fastq.gz
mv SRR5094400.fastq.gz SRR5094400_16_L003_R1_001.fastq.gz
mv SRR5094401.fastq.gz SRR5094401_17_L003_R1_001.fastq.gz
mv SRR5094404.fastq.gz SRR5094404_18_L003_R1_001.fastq.gz
mv SRR5094405.fastq.gz SRR5094405_19_L003_R1_001.fastq.gz
mv SRR5094408.fastq.gz SRR5094408_20_L003_R1_001.fastq.gz
mv SRR5094415.fastq.gz SRR5094415_21_L003_R1_001.fastq.gz
mv SRR5094419.fastq.gz SRR5094419_22_L003_R1_001.fastq.gz
mv SRR5094420.fastq.gz SRR5094420_23_L003_R1_001.fastq.gz
mv SRR5094421.fastq.gz SRR5094421_24_L003_R1_001.fastq.gz
mv SRR5094424.fastq.gz SRR5094424_25_L003_R1_001.fastq.gz
mv SRR5094428.fastq.gz SRR5094428_26_L003_R1_001.fastq.gz
mv SRR5094432.fastq.gz SRR5094432_27_L003_R1_001.fastq.gz
mv SRR5094436.fastq.gz SRR5094436_28_L003_R1_001.fastq.gz
mv SRR5094437.fastq.gz SRR5094437_29_L003_R1_001.fastq.gz
mv SRR5094438.fastq.gz SRR5094438_30_L003_R1_001.fastq.gz
mv SRR5094440.fastq.gz SRR5094440_31_L003_R1_001.fastq.gz
mv SRR5094443.fastq.gz SRR5094443_32_L003_R1_001.fastq.gz
mv SRR5094445.fastq.gz SRR5094445_33_L003_R1_001.fastq.gz
mv SRR5094449.fastq.gz SRR5094449_34_L003_R1_001.fastq.gz
mv SRR5094460.fastq.gz SRR5094460_35_L003_R1_001.fastq.gz
mv SRR5094469.fastq.gz SRR5094469_36_L003_R1_001.fastq.gz
mv SRR5094471.fastq.gz SRR5094471_37_L003_R1_001.fastq.gz
mv SRR5094472.fastq.gz SRR5094472_38_L003_R1_001.fastq.gz
mv SRR5094475.fastq.gz SRR5094475_39_L003_R1_001.fastq.gz
mv SRR5094478.fastq.gz SRR5094478_40_L003_R1_001.fastq.gz
mv SRR5094479.fastq.gz SRR5094479_41_L003_R1_001.fastq.gz
mv SRR5094480.fastq.gz SRR5094480_42_L003_R1_001.fastq.gz
mv SRR5094485.fastq.gz SRR5094485_43_L003_R1_001.fastq.gz
mv SRR5094489.fastq.gz SRR5094489_44_L003_R1_001.fastq.gz
mv SRR5094515.fastq.gz SRR5094515_45_L003_R1_001.fastq.gz
mv SRR5094517.fastq.gz SRR5094517_46_L003_R1_001.fastq.gz
mv SRR5094525.fastq.gz SRR5094525_47_L003_R1_001.fastq.gz
mv SRR5094528.fastq.gz SRR5094528_48_L003_R1_001.fastq.gz
mv SRR5094531.fastq.gz SRR5094531_49_L003_R1_001.fastq.gz
mv SRR5094532.fastq.gz SRR5094532_50_L003_R1_001.fastq.gz
mv SRR5094534.fastq.gz SRR5094534_51_L003_R1_001.fastq.gz
mv SRR5094536.fastq.gz SRR5094536_52_L003_R1_001.fastq.gz
mv SRR5094551.fastq.gz SRR5094551_53_L003_R1_001.fastq.gz
mv SRR5094552.fastq.gz SRR5094552_54_L003_R1_001.fastq.gz
mv SRR5094554.fastq.gz SRR5094554_55_L003_R1_001.fastq.gz
mv SRR5094565.fastq.gz SRR5094565_56_L003_R1_001.fastq.gz
mv SRR5094566.fastq.gz SRR5094566_57_L003_R1_001.fastq.gz
mv SRR5094569.fastq.gz SRR5094569_58_L003_R1_001.fastq.gz
mv SRR5094574.fastq.gz SRR5094574_59_L003_R1_001.fastq.gz
mv SRR5094576.fastq.gz SRR5094576_60_L003_R1_001.fastq.gz


#conda activate qiime2-amplicon-2024.10
#conda config --env --set subdir osx-64



#make output directory for qiime import
mkdir $wkdir/01_qiime_import_fastq/


#import reads into Qiime2
qiime tools import \
  --type 'SampleData[SequencesWithQuality]' \
  --input-path /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/00_reads/ \
  --input-format CasavaOneEightSingleLanePerSampleDirFmt \
  --output-path /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/01_qiime_import_fastq/QTI_reads.qza


/Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/02_qiime_dada2/



mkdir $wkdir/02_qiime_dada2/

#run dada2
qiime dada2 denoise-single --i-demultiplexed-seqs /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/01_qiime_import_fastq/QTI_reads.qza --o-table /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/02_qiime_dada2/table.qza --o-representative-sequences /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/02_qiime_dada2/rep-seqs.qza --o-denoising-stats /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/02_qiime_dada2/denoising-stats.qza --p-trunc-len 150





#download reference taxonomy for taxon assignment


mkdir $wkdir/03_qiime_reference_taxonomy/
cd $wkdir/03_qiime_reference_taxonomy/
wget https://data.qiime2.org/2022.2/common/silva-138-99-seqs-515-806.qza
wget https://data.qiime2.org/2022.2/common/silva-138-99-tax-515-806.qza



##run feature classifier to assign taxonomy to OTU's


mkdir $wkdir/04_qiime_feature-classifier_classify-consensus-vsearch/

qiime feature-classifier classify-consensus-vsearch \
  --i-query /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/02_qiime_dada2/rep-seqs.qza \
  --i-reference-reads /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/03_qiime_reference_taxonomy/silva-138-99-seqs-515-806.qza \
  --i-reference-taxonomy /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/03_qiime_reference_taxonomy/silva-138-99-tax-515-806.qza \
  --o-classification /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/04_qiime_feature-classifier_classify-consensus-vsearch/taxonomy.qza --o-search-results /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/04_qiime_feature-classifier_classify-consensus-vsearch/hits.qza






##Generate a tree for phylogenetic diversity analyses


mkdir $wkdir/05_qiime_phylogeny_align-to-tree-mafft-fasttree/


qiime phylogeny align-to-tree-mafft-fasttree \
  --i-sequences /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/02_qiime_dada2/rep-seqs.qza \
  --o-alignment /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/05_qiime_phylogeny_align-to-tree-mafft-fasttree/aligned-rep-seqs.qza \
  --o-masked-alignment /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/05_qiime_phylogeny_align-to-tree-mafft-fasttree/masked-aligned-rep-seqs.qza \
  --o-tree /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis//05_qiime_phylogeny_align-to-tree-mafft-fasttree/unrooted-tree.qza \
  --o-rooted-tree /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis//05_qiime_phylogeny_align-to-tree-mafft-fasttree/rooted-tree.qza






##Convert .qza files to file formats that I can actually use (csv, tsv, nwk)

mkdir $wkdir/06_qiime_export_dada2/

cd $wkdir/02_qiime_dada2/

qiime tools export \
  --input-path table.qza \
  --output-path /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis//06_qiime_export_dada2/exported-table




#make the feature table readable
cd $wkdir/06_qiime_export_dada2/exported-table
biom convert -i feature-table.biom -o feature-table.tsv --to-tsv

sed -e 's/\t/,/g' feature-table.tsv > feature-table.csv


    #feature-table.tsv was converted to feature-table.csv ; this is the feature table used for downstream analyses.

#get usable nwk files

mkdir $wkdir/05_qiime_phylogeny_align-to-tree-mafft-fasttree/export/
cd $wkdir/05_qiime_phylogeny_align-to-tree-mafft-fasttree/
for i in *; do qiime tools export --input-path $i --output-path /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis//05_qiime_phylogeny_align-to-tree-mafft-fasttree/export/$i; done
	#the rooted tree tree.nwk was renamed to tree.tree for phyloseq and used for downstream analyses

cd $wkdir/05_qiime_phylogeny_align-to-tree-mafft-fasttree/export/rooted-tree.qza/

cp tree.nwk  tree.tree



#export taxonomy
mkdir $wkdir/04_qiime_feature-classifier_classify-consensus-vsearch/export/
cd $wkdir/04_qiime_feature-classifier_classify-consensus-vsearch/
qiime tools export --input-path taxonomy.qza --output-path /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/04_qiime_feature-classifier_classify-consensus-vsearch/export/
  #taxonomy.tsv was used for downstream analyses




####amplicontraits 8-11-2025


conda activate qiime2-amplicon-2024.10
conda config --env --set subdir osx-64


qiime tools export --input-path /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/02_qiime_dada2/rep-seqs.qza --output-path /Users/gavin/genome/microbiology_spectrum_revisions_16S_2025/elegans_included_analysis/02_qiime_dada2/rep-seqs_fa


